### PR TITLE
imgproc: optimize corner detection series using AVX instructions

### PR DESCRIPTION
**Pull request by tomoaki0705**
### This pullrequest changes
 - use AVX intrinsic in corner detection series
 - measurement result will follow shortly

### CODE REVIEW

**Comment by tomoaki0705** [comment]
### Windows 7 + VS2012 (Corei7 3770 3.4GHz)
- performance measurement result extracted from ```opencv_perf_imgproc``` tests
  - ```Img_BlockSize_ApertureSize_k_BorderType_cornerHarris.cornerHarris/53```
  - ```Img_BlockSize_ApertureSize_BorderType_cornerEigenValsAndVecs.cornerEigenValsAndVecs/7```
  - ```Img_BlockSize_ApertureSize_BorderType_cornerMinEigenVal.cornerMinEigenVal/17```
  - numbers are median value in ```ms```
- I had to  manually define ```CV_AVX``` as 1 and also chane the Visual Studio Option to use ```/arch:AVX```
  - i.e.: I confirmed that AVX intrinsics are used
- ```before``` in the table below stands for universal intrinsic (SSE2) version

|condition|before|after|
|---------|------|-----|
|cornerHarris/53 | 3.54| 2.29|
|cornerEigenValsAndVecs/7 | 37.18| 24.54|
|cornerMinEigenVal/17 | 3.04| 2.05|

### discussions
 - I added some [```static inline``` functions](https://github.com/opencv/opencv/pull/8072/files#diff-b87906a65f2ae889cbba80fdab8491a6R54) which could be in better place.
 - I just couldn't figure out the best place so I placed it in the same file
 - I'm happy to have discussion of where to add those functions

File: modules/imgproc/src/corner.cpp [hunk 1]
```diff
@@ -67,7 +140,22 @@ static void calcMinEigenVal( const Mat& _cov, Mat& _dst )
...
+                __m256 v_a, v_b, v_c, v_t;
+                load_deinterleave(cov + j*3, v_a, v_b, v_c);
+                v_a = _mm256_mul_ps(v_a, half);
+                v_c = _mm256_mul_ps(v_c, half);
+                v_t = _mm256_sub_ps(v_a, v_c);
+                v_t = _mm256_add_ps(_mm256_mul_ps(v_b, v_b), _mm256_mul_ps(v_t, v_t));
+                _mm256_storeu_ps(dst + j, _mm256_sub_ps(_mm256_add_ps(v_a, v_c), _mm256_sqrt_ps(v_t)));
+            }
+        }
+#elif CV_SIMD128
```
Comment (alalek) [inline modules/imgproc/src/corner.cpp]: Why is "#elif" here?
If AVX is not available during runtime (but we compile with AVX), then we probably want to try SSE instructions too ("simd" variable should be spitted too).

Comment (tomoaki0705) [inline modules/imgproc/src/corner.cpp]: Thank you for your comment.
True, I agree your suggestion.
The only reason I added ```elif``` is that I followed same procedure from #7863 , but probably I should fix both.

**Comment by alalek** [comment]
Thank you! :+1:

**Pull request merged**